src/main/java/org/sonar/plugins/checkstyle/CheckstyleRulesDefinition.java

```diff
@@ -61,14 +63,30 @@ static void extractRulesData(NewRepository repository, String xmlRulesFilePath,
                 .getResourceAsStream(xmlRulesFilePath)) {
             ruleLoader.load(repository, resource, "UTF-8");
         }
-        ExternalDescriptionLoader.loadHtmlDescriptions(repository, htmlDescriptionFolder);
+        loadHtmlDescriptions(repository, htmlDescriptionFolder);
         try (InputStream resource = CheckstyleRulesDefinition.class
                 .getResourceAsStream("/org/sonar/l10n/checkstyle.properties")) {
             loadNames(repository, resource);
         }
         SqaleXmlLoader.load(repository, "/com/sonar/sqale/checkstyle-model.xml");
     }
 
+    private static void loadHtmlDescriptions(NewRepository repository, String htmlDescriptionFolder) {
+        // code adapted from:
+        // https://github.com/SonarSource/sslr-squid-bridge/blob/2.7.0.377/
+        // src/main/java/org/sonar/squidbridge/rules/ExternalDescriptionLoader.java
```

Comment:
Can we move these comments to the Javadoc?
Maybe copy the first sentence from sonar's javadoc (if there is one) and append a note saying it was copied from and use `{@link ExternalDescriptionLoader#loadHtmlDescriptions(NewRepository, String)}` instead of url to github.

Reply:
Yes, but should we also move https://github.com/checkstyle/sonar-checkstyle/blob/master/src/main/java/org/sonar/plugins/checkstyle/CheckstyleRulesDefinition.java#L73 to the Javadoc as it follows the same.

However, I don't quite understand what you mean with `{@link ExternalDescriptionLoader#loadHtmlDescriptions(NewRepository, String)}`?
Because for now, this class still exists, just like `loadNames` (see link before) still existed until they removed some of their deprecated code. I assume this will happen here as well, meaning `ExternalDescriptionLoader` won't exist at some point anymore.

Both methods don't have any Javadoc in `sslr-squid-bridge`.

Reply:
> shouldn't we also move ... to the Javadoc as it follows the same?

Yes I like that.

> Because for now, this class still exists, just like loadNames (see link before) still existed until they removed some of their deprecated code. I assume this will happen here as well, meaning ExternalDescriptionLoader won't exist at some point anymore.

Ok, URL has tag in it so it will never go bad unless github goes away.
`{@link` was just a javadoc tag that would take user to that method if they clicked on it like in an IDE. If class/method does go away, javadoc tool should print an error on the `{@link` as it won't point to a valid class.

I was mainly just pointing its usage since I feel it is a better way to point to a class/method in a javadoc. If everyone wants to stay with URL, I am fine.

Reply:
@muhlba91 , it this discussion point resolved ?
If yes, please pin @rnveach to approve PR.

Reply:
It has not been resolved.

Reply:
For this point I was waiting for a decision if we want to put dead Javadoc links in or stick with GitHub URL references?
I'm good with both ways but we should still keep the URL somewhere as well for a full reference to where the code was taken from (incl. `sslr-squid-bridge` version). Otherwise, you won't really know from where/when the code was taken over and whether it was modified by us already or not.

Reply:
I see no reason in reference of javadoc in other project, especially to non-existing code (non-existing javadoc).

Please move link from single comment to  loadHtmlDescriptions javadoc styles comment. Please do the same in "loadNames". Simply move whole text of comment to javadoc (no extra descriptions are required.)